wit/couchbase/document.wit

```diff
@@ -0,0 +1,23 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// Support CRUD access and modification of Documents stored in a Couchbase cluster.
+///
+/// Reference: https://docs.couchbase.com/c-sdk/current/concept-docs/documents.html
+interface document-crud {
+  use types.{document, document-id, document-error, mutation-metadata};
+
+  /// Insert a document with a new ID
+  insert: func(id: document-id, doc: document) -> result<mutation-metadata, document-error>;
+
+  /// Replace a document with the given ID with a new document
+  replace: func(id: document-id, doc: document) -> result<mutation-metadata, document-error>;
+
+  /// Create or update (replace) an existing document with the given ID
+  upsert: func(id: document-id, doc: document) -> result<mutation-metadata, document-error>;
+
+  /// Retrieve a document by ID
+  get: func(id: document-id) -> result<document, document-error>;
```

Comment:
Any particular reason to why the `CAS` not included in the result ?

Reply:
Hey you're right here -- we should absolutely contain the CAS -- the Golang SDK makes it available so we can here as well. I'll add a `document-get-result` or something of that sort here.

- [ ] Add `document-get-result` with `revision`, `expires-at`, and teh actual document
- [ ] Add options for getting a document

Reply:
@vados-cosmonic did you consider `timeout` and `use-replica` ?

Reply:
Yup! `document-get-options` looks like this now:

```wit
  /// Options for performing a document get
  record document-get-options {
    /// Whether expiry should also be retreived
    with-expiry: bool,

    /// Perform a projection of only specific paths of the result, returning only
    /// the fields indicated by the given list of paths.
    project: option<list<string>>,

    /// Timeout that should be used, in nanoseconds
    /// If not specified, the implementer *may* provide a default timeout.
    timeout-ns: option<u64>,

    /// How and whether to retry the operation
    retry-strategy: option<retry-strategy>,

    /// A known span to associate this lookup with
    parent-span: option<request-span>,
  }
```

Reply:
Oh but `use-replica` is actually not an option on [`GetOptions`](https://pkg.go.dev/github.com/couchbase/gocb/v2#GetOptions) as far as I can see -- should it be?

Reply:
> Oh but `use-replica` is actually not an option on [`GetOptions`](https://pkg.go.dev/github.com/couchbase/gocb/v2#GetOptions) as far as I can see (going off of [`Collection.Get`](https://pkg.go.dev/github.com/couchbase/gocb/v2#Collection.Get)) -- should it be?

No it does not but there is [`Collection.GetAnyReplica`](https://pkg.go.dev/github.com/couchbase/gocb/v2#Collection.GetAnyReplica). It was just a nice way to keep the interface light. Also because `useReplica` exists when using n1ql. So, consistency I guess ?

Reply:
Ah thanks for pointing that out -- in that case I can definitely add it -- providers can negotiate that option fairly easily then.

- [ ] add `use-replica` to document get